Sources/TuistGenerator/Linter/SettingsLinter.swift

```diff
@@ -67,11 +67,10 @@ final class SettingsLinter: SettingsLinting {
         )
 
         switch deploymentTarget {
-        case .iOS: if platform != .iOS { return [issue] }
-        case .macOS: if platform != .macOS { return [issue] }
         case .watchOS: if platform != .watchOS { return [issue] }
-        case .tvOS: if platform != .tvOS { return [issue] }
+        default: break
         }
```

Comment:
Can you explain this?

Reply:
I removed check for ios/macos/tvOS because each platform from [ios, macOS, tvOS] can support any combination from [ios, macOS, tvOS]. For example: Platform can be  tvOS but it can support only macOS. **SDKROOT = appletvos SUPPORTED_PLATFORMS = macosx.**

Reply:
Thanks for the additional info @dpliushchaiIOS - while that is possible in Xcode, would that result in a valid building target? Could the lint rules perhaps relax only when detecting valid multi platform combinations rather than none at all? I worry relaxing them here would prevent catching simple case errors which may be more common than the advanced multi platform use cases.

Reply:
@kwridan  It would result in a valid building target, you can check in my example app that I added to fixtures/. I am still not sure about .watchOS, still needs some testing but it seems to be possible to have a valid building target for any platform & any deploymentTarget. I mean that it might be possible to have target where SDKROOT is .watchOS but we can launch it on iphone. I am not sure how useful it will be for app but it is really powerfull for frameworks.